user_management/ui/views.py

```diff
@@ -20,16 +19,23 @@ class VerifyUserEmailView(VerifyAccountViewMixin, generic.RedirectView):
     As a RedirectView, this will return a HTTP 302 to LOGIN_URL on success.
     """
     permanent = False
+    already_verified = False
     url = settings.LOGIN_URL
     success_message = _('Your email address was confirmed.')
+    already_verified_message = _('Your email is already confirmed.')
     invalid_exception_class = InvalidExpiredToken
-    permission_denied_class = PermissionDenied
+    permission_denied_class = AlreadyVerifiedException
 
     def dispatch(self, request, *args, **kwargs):
-        self.verify_token(request, *args, **kwargs)
+        try:
+            self.verify_token(request, *args, **kwargs)
+        except self.permission_denied_class:
+            self.already_verified = True
+            self.success_message = self.already_verified_message
         return super(VerifyUserEmailView, self).dispatch(request, *args, **kwargs)
 
     def get(self, request, *args, **kwargs):
-        self.activate_user()
+        if not self.already_verified:
+            self.activate_user()
         messages.success(request, self.success_message)
```

Comment:
`success_message` gets set to `already_verified_message` ('Your email is already confirmed.') if they are 
`already_verified`